jina/docker/hubio.py

```diff
@@ -172,37 +172,34 @@ def push(self, name: str = None, readme_path: str = None, build_result: Dict = N
         """
         name = name or self.args.name
 
-        try:
-            # check if image exists
-            # fail if it does
-            if self._image_version_exists(
-                    build_result['manifest_info']['name'],
-                    build_result['manifest_info']['version'],
-                    jina_version
-            ):
-                raise Exception(f'Image with name {name} already exists. Will NOT overwrite.')
+        # check if image exists
+        # fail if it does
+        if self.args.no_overwrite and self._image_version_exists(
+                build_result['manifest_info']['name'],
+                build_result['manifest_info']['version'],
+                jina_version
+        ):
+            raise Exception(f'Image with name {name} already exists. Will NOT overwrite.')
+        else:
+            self.logger.debug(f'Image with name {name} does not exist. Pushing now...')
+        self._push_docker_hub(name, readme_path)
+
+        if not build_result:
+            file_path = get_summary_path(name)
+            if os.path.isfile(file_path):
+                with open(file_path) as f:
+                    build_result = json.load(f)
             else:
-                self.logger.debug(f'Image with name {name} does not exist. Pushing now...')
-            self._push_docker_hub(name, readme_path)
-
-            if not build_result:
-                file_path = get_summary_path(name)
-                if os.path.isfile(file_path):
-                    with open(file_path) as f:
-                        build_result = json.load(f)
-                else:
-                    self.logger.error(f'can not find the build summary file.'
-                                      f'please use "jina hub build" to build the image first '
-                                      f'before pushing.')
-
-            if build_result:
-                if build_result.get('is_build_success', False):
-                    _register_to_mongodb(logger=self.logger, summary=build_result)
-                if build_result.get('details', None) and build_result.get('build_history', None):
-                    self._write_slack_message(build_result, build_result['details'], build_result['build_history'])
+                self.logger.error(f'can not find the build summary file.'
+                                  f'please use "jina hub build" to build the image first '
+                                  f'before pushing.')
+
+        if build_result:
+            if build_result.get('is_build_success', False):
+                _register_to_mongodb(logger=self.logger, summary=build_result)
+            if build_result.get('details', None) and build_result.get('build_history', None):
+                self._write_slack_message(build_result, build_result['details'], build_result['build_history'])
 
-        except Exception as ex:
```

Comment:
why did we remove the `try..except` here?

Reply:
because the `raise` wasn't propagated. It needs to be propagated in order to be caught and dealt with by the handler. For now it's only used in the test, with `pytest.raises ...`

Reply:
We need to find a different way to propagate the raise without removing the overall `Exception` as this is one of the errors that signifies an error in hub-builder.